sdcm/cluster.py

```diff
@@ -3336,23 +3340,42 @@ def _create_session(self, node, keyspace, user, password, compression,
     def cql_connection(self, node, keyspace=None, user=None,  # pylint: disable=too-many-arguments
                        password=None, compression=True, protocol_version=None,
                        port=None, ssl_opts=None, connect_timeout=100, verbose=True):
-        node_ips = self.get_node_cql_ips()
-        wlrr = WhiteListRoundRobinPolicy(node_ips)
+        if connection_bundle_file := node.parent_cluster.connection_bundle_file:
+            wlrr = None
+            node_ips = []
+        else:
+            node_ips = self.get_node_cql_ips()
+            wlrr = WhiteListRoundRobinPolicy(node_ips)
         return self._create_session(node=node, keyspace=keyspace, user=user, password=password,
                                     compression=compression, protocol_version=protocol_version,
                                     load_balancing_policy=wlrr, port=port, ssl_opts=ssl_opts, node_ips=node_ips,
-                                    connect_timeout=connect_timeout, verbose=verbose)
+                                    connect_timeout=connect_timeout, verbose=verbose,
+                                    connection_bundle_file=connection_bundle_file)
 
-    def cql_connection_exclusive(self, node, keyspace=None, user=None,  # pylint: disable=too-many-arguments
+    def cql_connection_exclusive(self, node, keyspace=None, user=None,  # pylint: disable=too-many-arguments,too-many-locals
                                  password=None, compression=True,
                                  protocol_version=None, port=None,
                                  ssl_opts=None, connect_timeout=100, verbose=True):
-        node_ips = [node.cql_ip_address]
-        wlrr = WhiteListRoundRobinPolicy(node_ips)
+        if connection_bundle_file := node.parent_cluster.connection_bundle_file:
+            # TODO: handle the case of multiple datacenters
+            bundle_yaml = yaml.safe_load(connection_bundle_file.open('r', encoding='utf-8'))
+            node_domain = None
+            for _, connection_data in bundle_yaml.get('datacenters', {}).items():
+                node_domain = connection_data.get('nodeDomain').strip()
+            assert node_domain, "didn't found nodeDomain in bundle"
```

Comment:
Would be useful to print also the filepath.

Reply:
fixed